bsdlib: version 0.6.2 (#151)
NordicPlayground/nrfxlib: bsdlib/doc/CHANGELOG.rst

All notable changes to this project are documented in this file.

bsdlib 0.6.2
************

* TLS session cache is again disabled by default due to missing support in modem firmware versions older than 1.1.1 included
* :cpp:func:`sendto()` will now set errno to ``NRF_EISCONN`` instead ``NRF_EINVAL`` when passing an address if the socket type is ``NRF_SOCK_STREAM``
* calling :cpp:func:`connect()` on a already connected socket will now properly return ``NRF_EISCONN`` instead of ``NRF_EBADF``

> * sockets with family ``NRF_AF_LTE`` must now be created with type ``NRF_SOCK_DGRAM``

Reviewer: Maybe this can be changed to proper validation of family, type, and protocol arguments to socket(). Should probably also mention that 0 is not accepted to any of the arguments as 0 is in some cases accepted on posix/bsd sockets to create default socket configuration. And was commonly used to create AT sockets before.

Author: Implicitly, 0 is not accepted if we say type MUST be DGRAM, so I don't change that. I can also add: 
- Improved validation of family, type, and protocol arguments in socket()

* setting a timeout :cpp:func:`recv()` larger than the maximum supported value will now properly return ``NRF_EDOM`` instead ``NRF_EINVAL``
* fixed an overflow in timeout computation
* operations on sockets that don't match the socket family will now return ``NRF_EAFNOSUPPORT`` instead of ``NRF_EINVAL``
* creating a socket when no sockets are available will now return ``NRF_ENOBUFS`` instead of ``NRF_ENOMEM``

> 

Reviewer: * Added added support for poll on connect,  MSG_DONTWAIT on sendto for IP sockets?
* Proper validation of supported flags on send/receive for protocols

Reviewer 2: Yes, MSG_DONTWAIT on sendto for IP sockets is coming soon.

Author: There is no added support for poll on connect, and non-blocking send is also not in this yes.
The flag validation I can add.

All notable changes to this project are documented in this file.

bsdlib 0.6.2
************

> * TLS session cache is again disabled by default due to missing support in modem firmware versions older than 1.1.1 included.

Author: Regarding first change: we can't remove "included" because neither 1.1.1 doesn't support TLS session cache. Currently there is no released modem FW out there that supports TLS session cache. @b-gent  how would you reword it then?

> * TLS session cache is again disabled by default due to missing support in modem firmware versions older than 1.1.1 included.

Reviewer 3: Ah, sorry, I misunderstood the usage of 'included' here. 

Using 'current' might be tricky because we never know when a user is going to read it, so probably it's better to specify a version.

I suggest: "TLS session cache is now disabled by default due to missing support in modem firmware version 1.1.1 and older."

> * TLS session cache is again disabled by default due to missing support in modem firmware versions older than 1.1.1 included.

Author: That is ok!

> * :cpp:func:`sendto()` will now set errno to ``NRF_EISCONN`` instead ``NRF_EINVAL`` when passing an address if the socket type is ``NRF_SOCK_STREAM``.

Author: ok

> * Calling :cpp:func:`connect()` on a already connected socket will now properly return ``NRF_EISCONN`` instead of ``NRF_EBADF``.

Author: ok

* Sockets with family ``NRF_AF_LTE`` must now be created with type ``NRF_SOCK_DGRAM``.

> * Setting a timeout :cpp:func:`recv()` larger than the maximum supported value will now properly return ``NRF_EDOM`` instead ``NRF_EINVAL``.

Reviewer 3: ```suggestion
* Setting the timeout :cpp:func:`recv()` to a larger than the maximum supported value now properly returns ``NRF_EDOM`` instead of ``NRF_EINVAL``.
```

Author: I'll also add an 'in' after word 'timeout'.
Setting the timeout **in** :cpp:func:`recv()` to a larger than the maximum supported value now properly returns ``NRF_EDOM`` instead of ``NRF_EINVAL``.

Reviewer 3: yes, thanks

* Fixed an overflow in timeout computation.

> * Operations on sockets that don't match the socket family will now return ``NRF_EAFNOSUPPORT`` instead of ``NRF_EINVAL``.

Reviewer 3: ```suggestion
* Operations on sockets that do not match the socket family now return ``NRF_EAFNOSUPPORT`` instead of ``NRF_EINVAL``.
```

Author: ok

> * Creating a socket when no sockets are available will now return ``NRF_ENOBUFS`` instead of ``NRF_ENOMEM``.

Reviewer 3: ```suggestion
* Creating a socket when no sockets are available now returns ``NRF_ENOBUFS`` instead of ``NRF_ENOMEM``.
```

Author: ok

* Improved validation of family, type, and protocol arguments in :cpp:func:`socket()`.

> * Improved validation of supported flags on :cpp:func:`send()` and :cpp:func:`recv()` for protocols

Reviewer 3: ```suggestion
* Improved validation of supported flags on :cpp:func:`send()` and :cpp:func:`recv()` for protocols.
```

Author: ok
